plotly/tools.py

```diff
@@ -1477,6 +1477,149 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def _make_colorscale(colors):
+        """
+        Makes a list of colors into a colorscale-acceptable form
+
+        For documentation regarding to the form of the output, see
+        https://plot.ly/python/reference/#mesh3d-colorscale
+        """
+        colorscale = []
+        diff = 1./(len(colors) - 1)
+
+        for j in range(len(colors)):
+            colorscale.append([j * diff, colors[j]])
+        return colorscale
```

Comment:
🐄 This could be more concisely written as follows (i think?):

```
scale = 1./(len(colors) - 1)
return [[i * scale, color] for i, color in enumerate(colors)]
```